crates/bevy_render/src/camera/camera.rs

```diff
@@ -36,14 +36,51 @@ use wgpu::Extent3d;
 pub struct Viewport {
     /// The physical position to render this viewport to within the [`RenderTarget`] of this [`Camera`].
     /// (0,0) corresponds to the top-left corner
-    pub physical_position: UVec2,
+    pub physical_position: AbsOrPercVec,
     /// The physical size of the viewport rectangle to render to within the [`RenderTarget`] of this [`Camera`].
     /// The origin of the rectangle is in the top-left corner.
-    pub physical_size: UVec2,
+    pub physical_size: AbsOrPercVec,
     /// The minimum and maximum depth to render (on a scale from 0.0 to 1.0).
     pub depth: Range<f32>,
 }
 
+/// A vector wrapper that represents either absolute value or value as a percentage.
+///
+/// For example, used in representing viewport position, either as absolute pixel value, or as a
+/// percentage of render target size.
+#[derive(Reflect, Debug, Clone, Serialize, Deserialize)]
+pub enum AbsOrPercVec {
+    Absolute(UVec2),
+    Percentage(Vec2),
+}
+
+impl Default for AbsOrPercVec {
+    fn default() -> Self {
+        Self::Absolute(Default::default())
+    }
+}
+
+impl AbsOrPercVec {
+    /// Returns percentage of provided vec or absolute value from self.
+    #[inline]
+    pub fn as_absolute(&self, of: UVec2) -> UVec2 {
+        match self {
+            AbsOrPercVec::Absolute(v) => *v,
+            AbsOrPercVec::Percentage(v) => (of.as_vec2() * *v).as_uvec2(),
+        }
+    }
+
+    /// Returns percentage of provided vec or absolute value.
+    /// Returns `None` if self is `Percentage` and argument is `None`
+    #[inline]
+    pub fn as_absolute_opt(&self, of_opt: Option<UVec2>) -> Option<UVec2> {
```

Comment:
```suggestion
    pub fn try_as_absolute(&self, of_opt: Option<UVec2>) -> Option<UVec2> {
```

A little clearer and more idiomatic.